nemo/collections/common/data/lhotse/nemo_adapters.py

```diff
@@ -457,7 +461,13 @@ def make_cut_with_subset_inmemory_recording(
         return cut
 
     # Otherwise, apply the memory optimization.
-    cut = cut.truncate(offset=offset, duration=duration, preserve_id=True)
+    try:
+        cut = cut.truncate(offset=offset, duration=duration, preserve_id=True)
+    except Exception as e:
+        raise RuntimeError(
+            f"Lhotse cut.truncate failed with offset={offset}, duration={duration}, recording={recording}: {e}"
+        )
```

Comment:
```suggestion
        ) from e
```

to provide more detail. Can you share what was the error? We should debug whether it's a data issue or code issue.

Reply:
it's because the manifest has "duration=0" but the actual audio has length > 0. It'll be better to provide the error details here to users know where went wrong

Reply:
+1, my point is use the `raise RuntimeError(...) from e` syntax to provide the original exception details as well.

Reply:
done~